src/main/java/org/isf/menu/rest/UserController.java

```diff
@@ -358,5 +381,147 @@ public ResponseEntity<List<LitePermissionDTO>> retrievePermissionsByUsername(@Pa
+		}
+		if (!userName.equals(SecurityContextHolder.getContext().getAuthentication().getName()) 
+						&& !SecurityContextHolder.getContext().getAuthentication().getName().equals(ADMIN)) {
+		    throw new OHAPIException(new OHExceptionMessage("Not allowed."));
+		}
+		UserSetting userSetting = userSettingManager.getUserSettingById(id);
+		UserSetting updated;
+		if (userSetting == null) {
+		    LOGGER.info("No user settings with id {}", id);
+	            throw new OHAPIException(new OHExceptionMessage("UserSetting doesn't exist."));
+		}
+		
+		UserSetting uSetting = userSettingMapper.map2Model(userSettingDTO);
+		updated  = userSettingManager.updateUserSetting(uSetting);
+		if (updated == null) {
+		    LOGGER.info("UserSetting is not updated!");
+		    throw new OHAPIException(new OHExceptionMessage("UserSetting not updated."));
+		}
+		LOGGER.info("UserSetting successfully updated!");
+		return ResponseEntity.ok(userSettingMapper.map2DTO(updated));
+	}
+	
+	/**
+	 * Retrieves an existing {@link UserSettingDTO} by id.
+	 * 
+	 * @param id - id of userSetting {@link UserSetting} .
+	 * @return {@link UserSettingDTO} if the UserSetting exist , null otherwise.
+	 * @throws OHServiceException.
+	 */
+	@GetMapping(value = "/users/settings/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserSettingDTO> getUserSettingById(@PathVariable(name = "id")int id) throws OHServiceException {
+		LOGGER.info("Retrieve the userSetting By id {}:", id);
+		UserSetting userSetting = userSettingManager.getUserSettingById(id);
+		if (userSetting == null) {
+		    LOGGER.info("No user settings with id {}", id);
+	            return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+		}
+		return ResponseEntity.ok(userSettingMapper.map2DTO(userSetting));
+	}
+	
+	/**
+	 * Retrieves an existing {@link UserSettingDTO} by user.
+	 * 
+	 * @param userName - the name of user.
+	 * @param configName - the name of the userSetting {@link UserSetting} . 
+	 * @return {@link UserSettingDTO} if the UserSetting exist , null otherwise.
+	 * @throws OHServiceException.
+	 */
+	@GetMapping(value = "/users/{userName}/settings/{configName}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserSettingDTO> getUserSettingByUser(@PathVariable(name = "userName") String userName, @PathVariable(name = "configName") String configName) throws OHServiceException {
+		LOGGER.info("Retrieve the userSetting By user {} and configName {}:", userName, configName);
+		List<UserSetting> userSettings = userSettingManager.getUserSettingByUserName(userName);
+		if (userSettings == null || userSettings.isEmpty()) {
+		    LOGGER.info("UserSetting not found!");
+		    throw new OHAPIException(new OHExceptionMessage("UserSetting not found."));
+		}
+		UserSetting userSetting = userSettingManager.getUserSettingByUserNameConfigName(userName, configName);
+		if (userSetting == null) {
```

Comment:
this is were you would use the `!isPresent()` if you make the change I suggested in CORE